src/components/CreateCommentForm.jsx

```diff
@@ -0,0 +1,68 @@
+import { useState, useContext } from "react";
+import { UserContext } from "../contexts/User";
+import { postCommentByArticleId } from "../api";
+
+export default function CreateComment({ article_id, setComments }) {
+  const [selected, setSelected] = useState(false);
+  const [commentBody, setCommentBody] = useState("");
+  const [isError, setIsError] = useState(false);
+  const { username } = useContext(UserContext);
+  return (
+    <form
+      onSubmit={(e) => {
+        e.preventDefault();
+        setIsError(false);
+        const reqBody = { body: commentBody, username };
+        const localRef = String(Math.random());
+        setComments((currentComments) => {
+          return [
+            { body: commentBody, author: username, localRef },
+            ...currentComments,
+          ];
+        });
+        postCommentByArticleId(article_id, reqBody).catch(() => {
+          setIsError(true);
+          setComments((currentComments) => {
+            const newComs = currentComments.filter((comment) => {
+              if (!comment.localRef) return true;
+              return !comment.localRef === localRef;
+            });
```

Comment:
Super creative & smart way of removing a comment if the post fails, which is something that is often glossed over. My only issue with it is that the result of Math.random() isn't guaranteed to be unique. 

(even though it's VERY unlikely to conflict, it's technically not the best thing to do for that reason. But for the context of this app, I think you can get away with it.)